src/rfc5322.rs

```diff
@@ -471,48 +471,25 @@ impl Rfc5322Builder {
     }
 
     pub fn emit_folded(&mut self, s: &str) {
-        let mut cur_len = 0;
-        let mut last_space = 0;
-        let mut last_cut = 0;
-
-        for (pos, c) in s.char_indices() {
-            match c {
-                ' ' => {
-                    last_space = pos;
-                }
-                '\r' => {
-                    cur_len = 0;
-                }
-                '\n' => {
-                    cur_len = 0;
-                }
-                _ => {}
-            }
-
-            cur_len += 1;
-            // We've reached our line length, so
-            if cur_len >= MIME_LINE_LENGTH && last_space > 0 {
-                // Emit the string from the last place we cut it to the
-                // last space that we saw
-                self.emit_raw(&s[last_cut..last_space]);
-                // ... and get us ready to put out the continuation
+        let words = s.split_whitespace();
+        let mut current = String::new();
+        
+        for word in words {
+            if current.len() + 1 + word.len() > MIME_LINE_LENGTH && !current.is_empty() {
+                // The current word doesn't fit into the next line anymore
+                // The "+ 1" is for the space (" ") before the next word
+                self.emit_raw(&current);
                 self.emit_raw("\r\n\t");
-
-                // Reset our counters
-                cur_len = 0;
-                last_cut = last_space + s[last_space..].chars().next().unwrap().len_utf8();
-                last_space = 0;
+                current = word.to_owned();
+            } else {
+                if !current.is_empty() {
+                    current += " "; // Separate the existing line and the new word with a space
```

Comment:
This will replace tabs and other whitespace in the original string with spaces, is this always correct? Should be ok in most cases I guess, but maybe there are some cases where preserving tabs and not replacing them with spaces is important?